### PR TITLE
[test] Use targeted region push in conjunction with repush

### PR DESCRIPTION
**Pull request by adamxchen**
## Summary

Maybe it doesn't worth a PR but we'd like to know whether targeted reigon push can work with repush and turned out our integ tests don't have that.

So, this PR adds this part so we can ensure this compatibility doesn't break.

Also tried to fix a flaky unit test..

## How was this PR tested?

GHCI.

## Does this PR introduce any user-facing changes?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

**Comment by adamxchen** [comment]
The `TestDisabledPartitionEnablerService` is flaky... We spin up a thread and check the mocked result immediately. Sometimes it'd pass but sometimes it'd fail as well.. I am working on a small fix for this.

File: services/venice-controller/src/test/java/com/linkedin/venice/controller/TestDisabledPartitionEnablerService.java [hunk 1]
```diff
@@ -30,11 +34,32 @@ public void testCleanupBackupVersion() throws Exception {
...
+
+  }
+
+  class DisabledPartitionEnablerServiceForTesting extends DisabledPartitionEnablerService {
+    public DisabledPartitionEnablerServiceForTesting(
+        VeniceHelixAdmin admin,
+        VeniceControllerMultiClusterConfig multiClusterConfig,
+        Time time) {
+      super(admin, multiClusterConfig, time);
+      this.cleanupThread = new Thread(new DisabledPartitionEnablerTaskForTesting(), "TestTask");
```
Comment (adamxchen) [inline services/venice-controller/src/test/java/com/linkedin/venice/controller/TestDisabledPartitionEnablerService.java:52]: This probably isn't very elegant but other ways I've tried would be disruptive. Plz lmk if we have better way to test such cases.

File: services/venice-controller/src/test/java/com/linkedin/venice/controller/TestDisabledPartitionEnablerService.java [hunk 2]
```diff
@@ -30,11 +35,32 @@ public void testCleanupBackupVersion() throws Exception {
     doReturn(true).when(controllerConfig).isEnableDisabledReplicaEnabled();
     doReturn(clusters).when(config).getClusters();
     TestMockTime time = new TestMockTime();
-    DisabledPartitionEnablerService service = new DisabledPartitionEnablerService(admin, config, time);
+    DisabledPartitionEnablerService service = new DisabledPartitionEnablerServiceForTesting(admin, config, time);
 
     service.startInner();
+    Assert.assertTrue(latch.await(1, TimeUnit.SECONDS));
     verify(admin, atLeast(1)).enableDisabledPartition(clusterName, "", true);
+    service.stopInner();
```
Comment (lluwm) [inline services/venice-controller/src/test/java/com/linkedin/venice/controller/TestDisabledPartitionEnablerService.java:43]: Not an expert on Mockito, do you think [VerificationWithTimeout](https://site.mockito.org/javadoc/current/index.html?org/mockito/verification/VerificationWithTimeout.html) help in this case?

Comment (adamxchen) [reply services/venice-controller/src/test/java/com/linkedin/venice/controller/TestDisabledPartitionEnablerService.java:43]: Yes! I think so! That probably require less code changes.. Let me incorporate that.

Comment (adamxchen) [reply services/venice-controller/src/test/java/com/linkedin/venice/controller/TestDisabledPartitionEnablerService.java]: Done! @lluwm

**Review by lluwm — APPROVED** [review]
LGTM and thank you for fixing this flaky one!

**Pull request merged**